### ISSUE TITLE
Brain of Confusion Dodge Removes All Buffs and Debuffs

### ISSUE THREAD

**Issue by tchravis** [Pryaxis/TShock#2799]
* TShock version: TShock 5
* TShock build number (if known):

#### Reproduction steps (if applicable)?
1. Equip Brain of Confusion
2. Chug some potions
3. Activate Brain of Confusion's dodge
4. ALL buff duration countdowns turn to 0s.
Console said: 
"GetDataHandlers: VERBOSE: GetDataHandlers HandlePlayerBuffList handled event and sent data tchravis"
when the Brain of Confusion activated and wiped all of my buffs.

#### Any stack traces or error messages (if known)?
`Unknown`
PUT SUPER LONG ERROR MESSAGES IN THE TICK MARKS
```
#### Any screenshots?

It seems like the Player Buff Handler is doing something wrong, the red circle is what was logged when the Brain of Confusion's dodge activated. 
Even the Potion Sickness Debuff was set to 0 seconds, and 'permanent' buffs like Happy also briefly hit 0s. 
![image](https://user-images.githubusercontent.com/116663950/200007325-4abda3ac-7792-44a8-91f0-383558a0bc44.png)

#### Any log messages from files that end in `.log` or `.txt`? What are the last 100 log messages from the server console?
[2022-11-04_09-52-42.log](https://github.com/Pryaxis/TShock/files/9939544/2022-11-04_09-52-42.log)

#### What plugins and what versions of those plugins are you running?
Crossplay, Chestroom, FindPermission, Freeze, House Regions, NPC Blocker, Region View, TCR Discord, Terraria Chat Relay, WorldRefill
Though I doubt that any of these plugins are affecting player buffs.

If I didn't provide any logs this issue, please close my issue immediately. I'm sorry for the inconvenience.

**Comment by drunderscore** [comment]
Can't seem to reproduce this. Are you consistently able to reproduce this with any buffs active?
If you could provide a video it may help, but as it stands it's not reproducible.

**Comment by tchravis** [comment]
> 

I can record a video of it happening as soon as possible. This leads me to think that it might be a plugin that's causing this, but I have no idea which plugins might affect buffs, let alone when Brain of Confusion activates. I think that it also has the ability to turn godmode off somehow, but I can't replicate that one as consistently.

**Comment by tchravis** [comment]
> Can't seem to reproduce this. Are you consistently able to reproduce this with any buffs active? If you could provide a video it may help, but as it stands it's not reproducible.

![image](https://user-images.githubusercontent.com/116663950/200162880-a0469d42-d2f8-4b7c-a286-257cd39977ef.png)
this is what the buffs/debuffs look like (0s) when BoC activates. 

Since it's not reproducible I'm checking plugins to see what might affect buffs again, but I still doubt plugins like "Terraria Chat Relay" has anything to do with this. If I were to guess, it might be Crossplay?

**Comment by sgkoishi** [comment]
Someone claims that reproducing this bug requires dealing crit damage after dodge, not confirmed.
Also, this might already be fixed in the latest 5.1.1 release for 1.4.4.8?
